packages/repository/src/errors/entity-not-found.error.ts

```diff
@@ -0,0 +1,40 @@
+// Copyright IBM Corp. 2018. All Rights Reserved.
+// Node module: @loopback/repository
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+import {Entity} from '../model';
+
+export class EntityNotFoundError<ID, Props extends object = {}> extends Error {
+  code: string;
+  entityName: string;
+  entityId: ID;
+
+  constructor(
+    entityOrName: typeof Entity | string,
+    entityId: ID,
+    extraProperties?: Props,
+  ) {
+    const entityName =
+      typeof entityOrName === 'string'
+        ? entityOrName
+        : entityOrName.modelName || entityOrName.name;
+
+    const quotedId = JSON.stringify(entityId);
+
+    super(`Entity not found: ${entityName} with id ${quotedId}`);
+
+    Error.captureStackTrace(this, this.constructor);
+
+    this.code = 'ENTITY_NOT_FOUND';
+    this.entityName = entityName;
+    this.entityId = entityId;
```

Comment:
Should we set this to `quotedId` or not?

Reply:
No. The purpose of `quotedId` is to have a value that's easy to parse in a string, to make it easy for readers to distinguish between e.g. a number `1` and a string `"1"`.

Error properties like `entityId` and `entityName` can be serialized in different ways in the HTTP response. When using JSON, the distinction between different types (`undefined`, `null`, a number, a string, an object/array) is preserved.